src/sync/generic_network_transport.hpp

```diff
@@ -0,0 +1,324 @@
+            return service_error_code::function_syntax_error;
+        else if (code == "FunctionInvalid")
+            return service_error_code::function_invalid;
+        else if (code == "IncomingWebhookNotFound")
+            return service_error_code::incoming_webhook_not_found;
+        else if (code == "IncomingWebhookAlreadyExists")
+            return service_error_code::incoming_webhook_already_exists;
+        else if (code == "IncomingWebhookDuplicateName")
+            return service_error_code::incoming_webhook_duplicate_name;
+        else if (code == "RuleNotFound")
+            return service_error_code::rule_not_found;
+        else if (code == "APIKeyNotFound")
+            return service_error_code::api_key_not_found;
+        else if (code == "RuleAlreadyExists")
+            return service_error_code::rule_already_exists;
+        else if (code == "AuthProviderDuplicateName")
+            return service_error_code::auth_provider_duplicate_name;
+        else if (code == "RestrictedHost")
+            return service_error_code::restricted_host;
+        else if (code == "APIKeyAlreadyExists")
+            return service_error_code::api_key_already_exists;
+        else if (code == "IncomingWebhookAuthFailed")
+            return service_error_code::incoming_webhook_auth_failed;
+        else if (code == "ExecutionTimeLimitExceeded")
+            return service_error_code::execution_time_limit_exceeded;
+        else if (code == "NotCallable")
+            return service_error_code::not_callable;
+        else if (code == "UserAlreadyConfirmed")
+            return service_error_code::user_already_confirmed;
+        else if (code == "UserNotFound")
+            return service_error_code::user_not_found;
+        else if (code == "UserDisabled")
+            return service_error_code::user_disabled;
+        else
+            return service_error_code::unknown;
+    }
+
+    const service_error_code m_code;
+    const std::string m_raw_code;
+};
+
+} // namespace error
+
+
+enum class Method {
+    get, post, patch
+};
+
+/**
+ * An HTTP request that can be made to an arbitrary server.
+ */
+struct Request {
+public:
+    /**
+     * The HTTP method of this request.
+     */
+    Method method;
+
+    /**
+     * The URL to which this request will be made.
+     */
+    std::string url;
+
+    /**
+     * The number of seconds that the underlying transport should spend on an HTTP round trip before failing with an
+     * error.
+     */
+    int timeout_ms;
+
+    /**
+     * The HTTP headers of this request.
+     */
+    std::map<std::string, std::string> headers;
+
+    /**
+     * The body of the request.
+     */
+    std::string body;
+};
+
+/**
+ * The contents of an HTTP response.
+ */
+struct Response {
+public:
+    /**
+     * The status code of the HTTP response.
+     */
+    int status_code;
+
+    /**
+     * The headers of the HTTP response.
+     */
+    std::map<std::string, std::string> headers;
```

Comment:
TIL: Response headers can return multiple values using comma, e.g. `Key: val1, val2, val2`
I assume we are supporting this here in that format? The reason I'm asking is that OkHttp represents header values in their own wrapper class and I need to reconstruct the output for this API.

Reply:
Hmm... I suppose it could be? I do not believe any headers from Stitch return like that.